src/DataSet/ObjectDataSet.php

```diff
@@ -154,20 +177,29 @@ private function canCache(): bool
 
     private function hasCacheItem(#[ExpectedValues(['rules', 'reflectionProperties'])] string $name): bool
     {
+        if ($this->cacheKey === null) {
+            return false;
+        }
+
         if (!array_key_exists($this->cacheKey, self::$cache)) {
             return false;
         }
 
         return array_key_exists($name, self::$cache[$this->cacheKey]);
     }
 
-    private function getCacheItem(#[ExpectedValues(['rules', 'reflectionProperties'])] string $name): array
+    private function getCacheItem(#[ExpectedValues(['rules', 'reflectionProperties'])] string $name): mixed
     {
+        /** @psalm-suppress PossiblyNullArrayOffset */
         return self::$cache[$this->cacheKey][$name];
     }
 
-    private function setCacheItem(#[ExpectedValues(['rules', 'reflectionProperties'])] string $name, array $rules): void
+    private function setCacheItem(#[ExpectedValues(['rules', 'reflectionProperties'])] string $name, mixed $value): void
```

Comment:
Don't quite agree with this one, because it require changes for adding other keys, even it's kind of more "local" cache implementation. But OK, let's make type hints more specific at this point.

Additional changes:

- Also reflected these changes in `getCacheItem()`.
- Make the type of `$rules` variable containing cached data more specific (found that it can be improved along the way).